src/sickle.ts

```diff
@@ -20,6 +20,62 @@ export function formatDiagnostics(diags: ts.Diagnostic[]): string {
       .join('\n');
 }
 
+// TypeScript has an API for JSDoc already, but it's not exposed.
+// https://github.com/Microsoft/TypeScript/issues/7393
+// For now we create types that are similar to theirs so that migrating
+// to their API will be easier.  See e.g. ts.JSDocTag and ts.JSDocComment.
+export interface JSDocTag {
+  tagName?: string;
+  parameterName?: string;
+  text: string;
+}
+export interface JSDocComment { tags: JSDocTag[]; }
+
+// getJSDocAnnotation parses JSDoc out of a comment string.
+// Returns null if comment is not JSDoc.
+export function getJSDocAnnotation(comment: string): JSDocComment {
+  // TODO(evanm): this is a pile of hacky regexes for now, because we
+  // would rather use the better TypeScript implementation of JSDoc
+  // parsing.  https://github.com/Microsoft/TypeScript/issues/7393
+  let match = comment.match(/^\/\*\*([\s\S]*)\*\/$/);
+  if (!match) return null;
+  comment = match[1].trim();
+  // Strip all the " * " bits from the front of each line.
+  comment = comment.replace(/^\s*\* /gm, '');
+  let lines = comment.split('\n');
+  let tags: JSDocTag[] = [];
+  for (let line of lines) {
+    let match = line.match(/^@(\S+) *(.*)/);
```

Comment:
you have two variables called `match`, matching different things. I don't feel strongly about it, but maybe `lineMatch`?